base/views.jl

```diff
@@ -112,13 +112,14 @@ julia> A
 macro view(ex)
     if Meta.isexpr(ex, :ref)
         ex = replace_ref_begin_end!(ex)
-        if Meta.isexpr(ex, :ref)
-            ex = Expr(:call, view, ex.args...)
-        else # ex replaced by let ...; foo[...]; end
-            @assert Meta.isexpr(ex, :let) && Meta.isexpr(ex.args[2], :ref)
-            ex.args[2] = Expr(:call, view, ex.args[2].args...)
+        if !Meta.isexpr(ex, :let)
+            # wrap in a let block to prevent accidentally redefining `view` if used on the LHS, e.g.
+            #   @view(A[x]) = 2
+            ex = Expr(:let, Expr(:block), ex)
         end
-        Expr(:&&, true, esc(ex))
+        @assert Meta.isexpr(ex.args[2], :ref)
+        ex.args[2] = :(Base.view($(ex.args[2].args...))
```

Comment:
Ok, now this won't work.  We can actually shadow `Base` in local scopes:

```Julia
julia> let Base = 1
           @view (1:2)[:]
       end
ERROR: type Int64 has no field view
```

Also missing a parens at the end of the line)

Reply:
It'll work if you only `esc` the arguments, not the whole expression.

Reply:
yeah, i was trying to avoid that since i didn't understand what the `let` block was doing.